File: create_simple_workflow.py
import requests
import json
import os
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Configuration
API_KEY = os.getenv("NEXT_PRIVATE_VAPI_API_KEY")
BASE_URL = "https://api.vapi.ai"
ASSISTANT_ID = "cfaa163c-4a47-471b-a39e-95c12d0cb738"

# Headers for API requests
headers = {
    "Authorization": f"Bearer {API_KEY}",
    "Content-Type": "application/json"
}

def create_workflow():
    """Create a simple real estate workflow"""
    print("Creating simple real estate workflow...")

    # Define a simple workflow based on the example format
    payload = {
        "nodes": [
            {
                "type": "start",
                "name": "Start",
                "prompt": "Hello, I'm a real estate assistant for a luxury property company in Dubai. I'd like to understand your property preferences to help you find the perfect home. Is this a good time to talk?"
            },
            {
                "type": "gather",
                "name": "GatherLocation",
                "variable": "location",
                "prompt": "Which area in Dubai are you interested in? For example, Dubai Marina, Downtown, Palm Jumeirah, etc."
            },
            {
                "type": "say",
                "name": "ThankYou",
                "prompt": "Thank you for sharing your preferences. I'll help you find properties in {{location}}. A real estate specialist will contact you soon with options."
            },
            {
                "type": "end",
                "name": "EndCall"
            }
        ],
        "name": "SimpleRealEstateWorkflow",
        "edges": [
            {
                "from": "Start",
                "to": "GatherLocation"
            },
            {
                "from": "GatherLocation",
                "to": "ThankYou"
            },
            {
                "from": "ThankYou",
                "to": "EndCall"
            }
        ]
    }

    print(f"Sending request to {BASE_URL}/workflow with payload...")
    response = requests.post(f"{BASE_URL}/workflow", headers=headers, json=payload)
    print(f"Response status code: {response.status_code}")

    if response.status_code != 200:
        print(f"Error creating workflow: {response.text}")
        return None

    workflow_data = response.json()
    print(f"Workflow created with ID: {workflow_data['id']}")
    return workflow_data

def assign_workflow_to_assistant(workflow_id):
    """Assign the workflow to the assistant"""
    print(f"Assigning workflow {workflow_id} to assistant {ASSISTANT_ID}...")

    # Update the assistant with the workflow ID
    update_payload = {
        "workflow_id": workflow_id
    }

    response = requests.patch(f"{BASE_URL}/assistants/{ASSISTANT_ID}",
                             headers=headers,
                             json=update_payload)

    if response.status_code != 200:
        print(f"Error updating assistant: {response.text}")
        return False

    print(f"Successfully assigned workflow to assistant {ASSISTANT_ID}")
    return True

def main():
    """Main function"""
    print("Starting simple workflow creation...")

    if not API_KEY:
        print("Error: NEXT_PRIVATE_VAPI_API_KEY environment variable not set")
        return

    # Create workflow
    workflow_data = create_workflow()

    if not workflow_data:
        print("Error: Failed to create workflow")
        return

    # Assign workflow to assistant
    success = assign_workflow_to_assistant(workflow_data['id'])

    if success:
        print("\n=== Workflow Setup Complete ===")
        print(f"Workflow ID: {workflow_data['id']}")
        print(f"Assistant ID: {ASSISTANT_ID}")
        print("\nNext steps:")
        print("1. Test the workflow by making a call using the assistant")
        print("2. Monitor the call in the Vapi dashboard")
    else:
        print("Workflow setup incomplete due to errors")

if __name__ == "__main__":
    main()

File: create_simple_workflow.js
const fetch = require('node-fetch');

// Configuration - Direct token inclusion
const API_KEY = "d1529b85-51d5-47c0-9332-a73d40f7d62b";  // Your Vapi private API key
const BASE_URL = "https://api.vapi.ai";
const ASSISTANT_ID = "cfaa163c-4a47-471b-a39e-95c12d0cb738";

// Headers for API requests
const headers = {
    "Authorization": `Bearer ${API_KEY}`,
    "Content-Type": "application/json"
};

async function createWorkflow() {
    console.log("Creating real estate lead qualification workflow...");
    
    // Define the workflow with the format that works
    const payload = {
        "nodes": [
            // Start node
            {
                "type": "start",
                "name": "Start"
            },
            
            // Greeting
            {
                "type": "say",
                "name": "Greeting",
                "prompt": "Hello, I'm a real estate assistant for a luxury property company in Dubai. I'd like to understand your property preferences to help you find the perfect home. Is this a good time to talk?"
            },
            
            // Gather location
            {
                "type": "gather",
                "name": "GatherLocation",
                "prompt": "Which area in Dubai are you interested in? For example, Dubai Marina, Downtown, Palm Jumeirah, etc."
            },
            
            // Gather property type
            {
                "type": "gather",
                "name": "GatherPropertyType",
                "prompt": "What type of property are you looking for? For example, apartment, villa, penthouse, etc."
            },
            
            // Gather budget
            {
                "type": "gather",
                "name": "GatherBudget",
                "prompt": "What's your budget in USD? You can give me a range or a specific amount."
            },
            
            // Thank you
            {
                "type": "say",
                "name": "ThankYou",
                "prompt": "Thank you for sharing your preferences. I'll help you find properties that match your criteria. A real estate specialist will contact you soon with options."
            },
            
            // End call
            {
                "type": "end",
                "name": "EndCall"
            }
        ],
        "name": "RealEstateLeadQualificationWorkflow",
        "edges": [
            {
                "from": "Start",
                "to": "Greeting"
            },
            {
                "from": "Greeting",
                "to": "GatherLocation"
            },
            {
                "from": "GatherLocation",
                "to": "GatherPropertyType"
            },
            {
                "from": "GatherPropertyType",
                "to": "GatherBudget"
            },
            {
                "from": "GatherBudget",
                "to": "ThankYou"
            },
            {
                "from": "ThankYou",
                "to": "EndCall"
            }
        ]
    };
    
    try {
        const response = await fetch(`${BASE_URL}/workflow`, {
            method: 'POST',
            headers: headers,
            body: JSON.stringify(payload)
        });
        
        if (!response.ok) {
            const errorText = await response.text();
            console.error(`Error creating workflow: ${errorText}`);
            return null;
        }
        
        const data = await response.json();
        console.log(`Workflow created with ID: ${data.id}`);
        return data;
    } catch (error) {
        console.error(`Error creating workflow: ${error.message}`);
        return null;
    }
}

async function main() {
    console.log("Starting workflow creation...");
    
    // Create workflow
    const workflowData = await createWorkflow();
    
    if (!workflowData) {
        console.error("Error: Failed to create workflow");
        return;
    }
    
    console.log("\n=== Workflow Setup Complete ===");
    console.log(`Workflow ID: ${workflowData.id}`);
    console.log(`Assistant ID: ${ASSISTANT_ID}`);
    console.log("\nNext steps:");
    console.log("1. Go to the Vapi dashboard (https://dashboard.vapi.ai)");
    console.log("2. Navigate to Assistants and select your assistant");
    console.log("3. Assign the workflow ID to your assistant");
    console.log("4. Test the workflow by making a call using the assistant");
}

// Run the main function
main().catch(error => {
    console.error("Error:", error);
});
